service/src/main/java/com/fecklessweasel/service/UserSessionUtil.java

```diff
@@ -0,0 +1,173 @@
+        // cookie is our only cookie. If we start saving more than five or ten
+        // we should use a different strategy for indexing and searching.
+        for (Cookie cookie : cookies) {
+            if (cookie.getName().equals(AUTH_COOKIE)) {
+                return UserSession.resumeFromSessionString(connection, cookie.getValue());
+            }
+        }
+
+        // No auth cookie present.
+        throw new ServiceException(ServiceStatus.NOT_AUTHENTICATED);
+    }
+
+    /**
+     * Resumes a session from the authentication cookie stored in a request.
+     * Connection to SQL is obtained automatically.
+     * @param request The HTTP request containing the authentication cookie.
+     * @throws ServiceException Thrown if the service encounters an exception.
+     * INVALID_SESSION_HEADER, ACCESS_DENIED, or NOT_AUTHENTICATED ServerStatus
+     * if unable to authenticate.
+     * @return The resumed UserSession.
+     */
+    public static UserSession resumeSession(final HttpServletRequest request)
+        throws ServiceException {
+
+        try {
+            // Open a SQL connection and create the user and log them in.
+            return SQLSource.interact(new SQLInteractionInterface<UserSession>() {
+                    @Override
+                    public UserSession run(Connection connection)
+                        throws ServiceException, SQLException {
+
+                        return resumeSession(connection,
+                                             request);
+                    }
+                });
+        } catch (ServiceException ex) {
+            if (ex.status == ServiceStatus.NOT_AUTHENTICATED) {
+                return null;
+            }
+
+            // Bubble up exception to user interface.
+            throw ex;
+        }
+    }
+
+    /**
+     * Ends all currently active sessions for the currently logged in user.
+     * @param request The HTTP request containing the auth cookie of the
+     * current user.
+     * @param response The HTTP response that will receive the auth cookie
+     * deletion.
+     * @throws ServiceException Thrown if the service encounters an error.
+     */
+    public static void endAllSessions(final HttpServletRequest request,
+                                      final HttpServletResponse response)
+        throws ServiceException {
+
+        // Open a SQL connection and create the user and log them in.
+        SQLSource.interact(new SQLInteractionInterface<Integer>() {
+                @Override
+                public Integer run(Connection connection)
+                    throws ServiceException, SQLException {
+
+                    UserSession session = resumeSession(connection,
+                                                        request);
+
+                    // Delete sessions in DB.
+                    UserSession.endAll(connection, session.getUser().getUsername());
```

Comment:
Do we want to delete? maybe just set an inactive flag instead. Probably out of scope for this ticket, but something to consider.